### ISSUE TITLE
Review names and document all output files and figures

### ISSUE THREAD

**Issue by ewallace** [riboviz/riboviz#395]
A reviewer said:

> Some of the names of the RiboViz2 output files are not intuitive. For example, a file labeled "3nt_periodicity.pdf" would indicate a plot showing the periodicity of reads. Instead, it appears to be a metagene plot which can indicate the periodicity but should include ‘metagene’ in the plot title as well. "pos_sp_rpf_norm_reads.pdf" appears to be a normalised metagene profile but the name here is very different. Consequently, clear explanations of the output files should be provided.

This is accurate and helpful, we need to respond to it:
* [x] go through all output files (tsv and figures) and write a description of them, then consider renaming.
* [x] new documentation file `docs/user/riboviz_outputs.md` that collects those descriptions of all the output files. 
* [ ] update output file names in `docs/user/prep-riboviz-operation.md` and link that to `docs/user/riboviz_outputs.md`
* [ ] update `prep_riboviz.nf` and/or `generate_stats.figs.R` and the static html to account for new names. Be prepared to grep the repository.
* [ ] update system tests to account for new file names.

The next step is to list the output files and make a checklist for all of them, further down this issue ticket.

Let's discuss with @FlicAnderson and @swinterbourne tomorrow.

**Comment by FlicAnderson** [comment]
So currently we have this documentation of the output files in [prep-riboviz-operation.md](https://github.com/riboviz/riboviz/blob/main/docs/user/prep-riboviz-operation.md#output-files).

**Comment by ewallace** [comment]
I created branch [output-docs-395](https://github.com/riboviz/riboviz/tree/output-docs-395) for this. About to start a new document in that branch, `docs/user/riboviz_outputs.md`.

**Comment by 3mma-mack** [comment]
Reviewing names, I think the ones that are okay as they are would be files like the bam and H5 files, which just have the sample name followed by the file type. I also think read_lengths.tsv and pdf is a pretty self explanatory name so probably doesn't need changed

**Comment by FlicAnderson** [comment]
Slightly related documentation table of contents issue ticket [here](https://github.com/riboviz/riboviz/issues/400) (as if #400 is resolved soon, then the new documentation for #395 should be added to it)

**Comment by 3mma-mack** [comment]
The ones I can remember confusing me based on their name when I started were; 3nt_periodicity, pos_sp_nt_freq, pos_sp_rpf_norm_reads, codon_ribodens, and startcodon_ribogridbar

**Comment by ewallace** [comment]
Checklist of output files. Let's tick them off as we rename or add to documentation

* [ ]`TPMs_collated.tsv`
* [ ]`read_counts.tsv`
* [ ]`<SAMPLE_ID>.bam`
* [ ]`<SAMPLE_ID>.bam.bai`
* [ ]`minus.bedgraph`
* [ ]`plus.bedgraph`
* [ ]`<SAMPLE_ID>.h5`
* [ ]`3nt_periodicity.tsv`
* [ ]`3nt_periodicity.pdf`
* [ ]`read_lengths.tsv`
* [ ]`read_lengths.pdf`
* [ ]`pos_sp_nt_freq.tsv`
* [ ]`pos_sp_rpf_norm_reads.pdf`
* [ ]`pos_sp_rpf_norm_reads.tsv`
* [ ]`features.pdf`
* [ ]`tpms.tsv`
* [ ]`codon_ribodens.tsv`
* [ ]`codon_ribodens.pdf`
* [ ]`startcodon_ribogridbar.pdf`
* [ ]`startcodon_ribogrid.pdf`
* [ ]`3ntframe_bygene.tsv`
* [ ]`3ntframe_propbygene.pdf`
* [ ]`<SAMPLE_ID>_output_report.html`

**Comment by FlicAnderson** [comment]
I think the following could be improved in naming, mentioning metagene plots where that's what's happening: 
* 3nt_periodicity  
* pos_sp_nt_freq  (expand)
* pos_sp_rpf_norm_reads  (expand, perhaps doesn't need RPF in it, since we don't currently output rna-seq position specific normalised reads?)
* features  (features of what could be clearer)
* tpms  (is this sufficiently known as an acronym? Perhaps include if 'by sample' or 'per gene'?) 
* codon_ribodens  (ribodens is not particularly intuitive?)
* startcodon_ribogridbar  (perhaps re-word to be more suggestive that this is a bar-chart of counts of reads for each in a specific set of read-lengths? Does 'startcodon' need to be included? @ewallace suggests yes, since this is a metagene of what's happening around the start codon, that should be included)  
* startcodon_ribogrid  (similar points as startcodon_ribogrid)
* 3ntframe_bygene  (could this be left as 'frame by gene'?)
* 3ntframe_propbygene  (expand to 'proportion_by_gene' perhaps? 'prop' isn't clear)
* TPMs_collated   (collated between samples in the dataset, so perhaps could indicate this?)
* read_counts (this is read counts at each stage of the riboviz workflow, for all samples)

**Comment by ewallace** [comment]
@3mma-mack agreed to take a crack at this to write documentation and suggest new names where needed. Then we will discuss.

**Comment by swinterbourne** [comment]
In terms of the tsv files - it could be useful for the documentation to contain a brief structure description (below) with a brief description of what each column contains (e.g explanation for abbreviations used) in addition to a general description of what the file is. 
```
gene	       Ct_fr0	Ct_fr1	Ct_fr2	pval_fr0vs1	pval_fr0vs2	pval_fr0vsboth
YAL068C	          0	  0	  0	   1	              1	               1
YAL067W-A	  0       0	  0	   1	              1	               1
```

**Comment by ewallace** [comment]
@swinterbourne agree. Also useful to mention the name of the function that produces it, where applicable?

**Comment by 3mma-mack** [comment]
I have added some first draft descriptions `riboviz_outputs.md` and have included tables showing the structure of tsv files. I still need to add descriptions for features.pdf, codon_ribodens.tsv, and codon_ribodens.pdf, as I am currently running the Brar dataset so I can see and properly describe these files. 

I'm also planning on replacing the tables I have in the files at the moment with those produced by the Brar dataset so the tables have consistent genes which will hopefully minimise any confusion. 

I also haven't added anything to Sample.bam, Sample.bam.bai, Minus.bedgraph, Plus.bedgraph yet as I was struggling to fund the py file that is actually used to create them and was not sure how to go about describing them.

**Comment by mikej888** [comment]
.bam and .bam.bai are created by samtools. .bedgraph is created by bedtools.

**Comment by 3mma-mack** [comment]
I have added short descriptions on how .bam, .bam.bai and the .bedgraph files are created. Additionally I have added a 'potential new file name' section to each file description, feel free to add to it.

I plan to add descriptions for the final files, update the tables and add images of graphs once the Brar dataset has finished running

**Comment by 3mma-mack** [comment]
I have a first draft of riboviz_outputs, which includes a description, and either an image or a table depending on whether the file is a  pdf or a tsv file. 

I plan to go through the documentation again to add more detailed descriptions on how each files is created.

**Comment by FlicAnderson** [comment]
https://github.com/riboviz/riboviz/blob/output-docs-395/docs/user/riboviz_outputs.md

**Comment by swinterbourne** [comment]
I had a quick look. I think it is coming along nicely, main thing that I think could be added would be some information in terms of quality control checks (e.g. look at read_lengths.pdf, check peak to see if adapter has been removed correctly). Either its own little QC section listing which files to you should check and what to look for or a sentence within the relevant sections describing a QC for the relevant pdfs.

**Comment by 3mma-mack** [comment]
Current brainstorm for new file names:

**TPMs_collated.tsv** : TPMs_all_samples, TPMs_samples_collated, TPMs_genes_collated, TPMs_samples_genes_compared, TPMs_all_genes_all_samples

**read_counts.tsv** : reads_per_process, reads_per_stage, reads_stage_inputs

**3nt_periodicity.tsv** : meta_read_counts_start_stop, meta_reads_start_stop, all_reads_meta_transcript, reads_meta_feature_transcript

**pos_sp_rpf_norm_reads.pdf** : norm_reads_transcript, norm_reads_start_stop

**pos_sp_rpf_norm_reads.tsv** : norm_reads_transcript, norm_reads_start_stop

**sequence_features.tsv** : sequence_features_vs_tpms, tpms_feature_changes, feature_trends_tpms

**features.pdf** : sequence_features_vs_tpms, tpms_feature_changes, feature_trends_tpms

**tpms.tsv** : <SAMPLE_ID>_tmps

**codon_ribodens.tsv** : APE_codondens, APE_feature_codonden, APE_codondens_trend

**codon_ribodens.pdf** : APE_codondens, APE_feature_codonden, APE_codondens_trend

**startcodon_ribogridbar.pdf** : startcodon_gridbar, start_gridbar_readlen, readlen_gridbar_start

**startcodon_ribogrid.pdf** : startcodon_readlen_heatmap, heatmap_readlen_pos

These are just suggestions so feel free to add to the lists. I haven't come up with suggestions for 3ntframe_bygene.tsv, 3ntframe_bygene.pdf, read_lengths.tsv, read_lengths.pdf, <SAMPLE_ID>.h5, plus.bedgraph, minus.bedgraph, <SAMPLE_ID>.bam.bai, <SAMPLE_ID>.bam, or <SAMPLE_ID>_output_report.html as I feel they are well named already but if anyone feels differently then feel free to add suggestions.

**Comment by ewallace** [comment]
Thinking about the `features.pdf` file, those are features of CDS or ORFs. So I think that `CDS_features` or `CDS_tpms_vs_features` might work?

**Comment by 3mma-mack** [comment]
The features are in ORFs so potentially ORF_features or ORF_tpms_vs_features

**Comment by mikej888** [comment]
To add to the list of changes arising from changing file names:

[ ] Update Python constants, variables or Nextflow channel names and variables that represent these files to be consistent with the new file names.

**Comment by 3mma-mack** [comment]
I have edited the document to include a small paragraph at the top of the document describing features of read_lengths.pdf and 3nt_periodicity.pdf to look for as a quality control measure when first running the riboviz pipeline, as suggested by @swinterbourne 

@ewallace would you be able to check the document when you have time. The document is at a point where I am not sure what to add, if I need to add anything else, or whether I should change the format so your input would be appreciated.

**Comment by 3mma-mack** [comment]
We have discussed some of the new names for files at the 11/08 developers meeting 
The names we have decided on so far are:

TPMs_collated.tsv  -> **TPMs_all_CDS_all_samples**
read_counts.tsv  -> **read_counts_per_file**
3nt_periodicity.tsv -> **metagene_start_stop_read_counts**
pos_sp_rpf_norm_reads.pdf -> **metagene_normalized_profile_start_stop**
pos_sp_rpf_norm_reads.tsv -> **metagene_normalized_profile_start_stop**
sequence_features.tsv -> **ORF_TPMs_vs_features**
features.pdf  -> **ORF_TPMs_vs_features**
tpms.tsv  -> **ORF_TPMs_and_counts**

We are planning on coming back to: codon_ribodens.tsv , codon_ribodens.pdf , startcodon_ribogridbar.pdf, and startcodon_ribogrid.pdf

Any input is welcome.

**Comment by FlicAnderson** [comment]
Just thinking about it, things like `3ntframe_bygene.*` ought to match our preference for 'ORF' over 'gene', and perhaps we can just drop '3nt' and use 'read_frame' instead?

**Comment by 3mma-mack** [comment]
Ah. Thanks for pointing that out, I'll have a look and add their descriptions to `riboviz-outputs.md`.

**Comment by 3mma-mack** [comment]
I have merged `develop` into `output-docs-395` and then `output-docs-395` into `update-file-names-411` so that there is a copy of riboviz_outputs.md on branch `update-file-names-411`. 

I plan to update riboviz_outputs.md so the names of files listed match the new file names where they have been changed

**Comment by 3mma-mack** [comment]
I plan to delete the branch and issue once file names have been decided for the following files:

* **codon_ribodens.tsv** - suggestions include: APE_codondens, APE_feature_codonden, APE_codondens_trend
* **codon_ribodens.pdf** - suggestions include: APE_codondens, APE_feature_codonden, APE_codondens_trend
* **startcodon_ribogridbar.pdf** - suggestions include: startcodon_gridbar, start_gridbar_readlen, readlen_gridbar_start
* **startcodon_ribogrid.pdf** - suggestions include: startcodon_readlen_heatmap, heatmap_readlen_pos
* **3ntframe_bygene.tsv** - suggestions include: readframe_byORF
* **3ntframe_bygene.pdf** - suggestions include: readframe_byORF

Once names have been decided then all of the next steps will be covered by issue #411 so this issue will be able to be closed

**Comment by FlicAnderson** [comment]
> I plan to delete the branch and issue once file names have been decided for the following files:
> 
> * **codon_ribodens.tsv** - suggestions include: APE_codondens, APE_feature_codonden, APE_codondens_trend
> * **codon_ribodens.pdf** - suggestions include: APE_codondens, APE_feature_codonden, APE_codondens_trend

I think 'ribodens' isn't particularly helpful, can we expand to 'ribosome_density_at_APE_sites' or similar?

**Comment by FlicAnderson** [comment]
> * **startcodon_ribogridbar.pdf** - suggestions include: startcodon_gridbar, start_gridbar_readlen, readlen_gridbar_start  
What about something like `start_codon_read_length_bar_chart`?

> * **startcodon_ribogrid.pdf** - suggestions include: startcodon_readlen_heatmap, heatmap_readlen_pos  
I think that this type of plot has been called a 'ribogrid' by the Lareau lab who I believe came up with this visualization? (correct me if I'm wrong  @lianafaye).  I think we can probably call it a ribogrid (with a bit in the name about what's plotted against what perhaps for extra clarity for the unfamilar?), if the documentation of outputs explains clearly that it's a type of heatmap plot and maybe includes a link to examples in @lianafaye 's paper?

**Comment by FlicAnderson** [comment]
> * **3ntframe_bygene.tsv** - suggestions include: readframe_byORF
> * **3ntframe_bygene.pdf** - suggestions include: readframe_byORF

I like the suggestion, but I think for consistency, we should probably snake_case_each_word? e.g: read_frame_by_ORF?

**Comment by ewallace** [comment]
I agree with @FlicAnderson: read_frame_per_ORF or frame_proportions_per_ORF?

codon_ribodens.tsv : norm_density_APEsites_per_codon ?

startcodon_ribogridbar.pdf : metagene_start_read_counts_by_read_length ? Certainly this

startcodon_ribogrid.pdf : startcodon_readlen_heatmap, heatmap_readlen_pos

???

**Comment by 3mma-mack** [comment]
Current name suggestions following discussions at the meeting and on this issue ticket. Unless there are any objections to them raised here or at the meeting I plan to power through in the name changes in and aim to have all file names changed by the end of the week. 

* **codon_ribodens.tsv** - norm_density_APEsites_per_codon.tsv
* **codon_ribodens.pdf** -  norm_density_APEsites_per_codon.pdf
*  **codon_ribodens_gathered.tsv** - norm_density_APEsites_per_codon_gathered.tsv
* **startcodon_ribogridbar.pdf** - metagene_start_barplot_bylength.pdf
* **startcodon_ribogrid.pdf** - metagene_start_ribogrid_bylength.pdf
* **3ntframe_bygene.tsv** - read_frame_per_ORF.tsv
* **3ntframe_bygene_filtered.tsv** - read_frame_per_ORF_filtered.pdf
* **3ntframe_propbygene.pdf** - frame_proportions_per_ORF.tsv
* **read_lengths.pdf** - read_counts_by_length.pdf
* **read_lengths.tsv** - read_counts_by_length.tsv
*  **pos_sp_nt_freq.tsv** - nt_freq_per_read_length_pos.tsv

**Comment by 3mma-mack** [comment]
> Current name suggestions following discussions at the meeting and on this issue ticket. Unless there are any objections to them raised here or at the meeting I plan to power through in the name changes in and aim to have all file names changed by the end of the week.
> 
> * **codon_ribodens.tsv** - normalized_density_APEsites_per_codon.tsv
> * **codon_ribodens.pdf** -  normalized_density_APEsites_per_codon.pdf
> * **codon_ribodens_gathered.tsv** - normalized_density_APEsites_per_codon_long.tsv
> * **startcodon_ribogridbar.pdf** - metagene_start_barplot_by_length.pdf
> * **startcodon_ribogrid.pdf** - metagene_start_ribogrid_by_length.pdf
> * **3ntframe_bygene.tsv** - read_frame_per_ORF.tsv
> * **3ntframe_bygene_filtered.tsv** - read_frame_per_ORF_filtered.pdf
> * **3ntframe_propbygene.pdf** - frame_proportions_per_ORF.pdf
> * **read_lengths.pdf** - read_counts_by_length.pdf
> * **read_lengths.tsv** - read_counts_by_length.tsv
> * **pos_sp_nt_freq.tsv** - nt_freq_per_read_position.tsv

New names accepted, so review of names can close. The rest of the work changing the files will be documented on #411

**Comment by FlicAnderson** [comment]
This work was completed on issue ticket #411, merged as part of PR #422 and the contents of branch `output-docs-395` are contained within `develop` via the 411 branch, so I'm deleting branch `output-docs-395` to tidy up.
